### PR TITLE
Make use_all config friendlier

### PR DESCRIPTION
**Pull request by cwjenkins**
Add new flag, `suppress_not_found`, to silence instrumentation that isn't found.
Separate out 'failed to install' from 'not found' in logs during install_all method

### CODE REVIEW

**Review by ahayworth — COMMENTED** [review]
Things I like: a more detailed error message! 🎉 
Things I'm not sure about: should this be an option we ask users to pass in? Or is there a better way to do it?

I think I'm wondering specifically if we shouldn't do:
- logger.info "installed successfully!"
- logger.debug "failed to install... #{reason}" kind of thing

Basically I don't know that anyone _ever_ has the full set of dependencies for `instrumentation-all` available. I presume that everyone would want to use this option (so maybe we should default this to `true` instead of `false`).

Thoughts?

**Comment by ahayworth** [comment]
The only thought that immediately comes to mind is "warn" is better than "error" I think, but ... I'm not sure otherwise. @open-telemetry/ruby-maintainers @open-telemetry/ruby-approvers any thoughts?

**Review by ahayworth — APPROVED** [review]
LGTM, thank you! Apologies for the delay, my work responsibilities have shifted so I haven't been super active around otel-ruby recently. I'm but a lowly approver on this repo, so we will need someone from @open-telemetry/ruby-maintainers to take a look.

**Comment by cheempz** [comment]
Upvote! Running into lots of warning messages on startup due to missing dependency and this change would be really useful :)

**Review by fbogsany — APPROVED** [review]
This LGTM. I'd like to get a 👍 from @robertlaurin before I merge - he's the expert on the Registry & instrumentation config code.

**Review by robbkidd — APPROVED** [review]
👍🏻 Big fan of more nuanced console messaging.

**Review by simi — APPROVED** [review]
:pray: LGTM

**Comment by cheempz** [comment]
Helloooo a plea/nag... could this be looked again by the maintainers please?

**Review by robertlaurin — APPROVED** [review]

**Review by simi — APPROVED** [review]

**Comment by cheempz** [comment]
New year, new ping on this ;).  Could this one be merged and released soon?  Our customer highlighted again as a friction point.

**Review by kaylareopelle — APPROVED** [review]

**Review by mwear — APPROVED** [review]

**Comment by ykitamura-mdsol** [comment]
is a new version of the opentelemetry-instrumentation-registry gem going to be released anytime soon?

**Pull request merged**
